acceptance/tests/connect/connect_helper.go

```diff
@@ -17,40 +17,83 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-const staticClientName = "static-client"
-const staticServerName = "static-server"
+const (
+	staticClientName = "static-client"
+	staticServerName = "static-server"
+)
 
-// ConnectInjectConnectivityCheck is a helper function used by the connect tests and cli smoke tests to test service
-// mesh connectivity.
-func ConnectInjectConnectivityCheck(t *testing.T, ctx environment.TestContext, cfg *config.TestConfig, secure bool, autoEncrypt bool, cli bool) {
-	helmValues := map[string]string{
-		"connectInject.enabled": "true",
+// ConnectHelper configures and runs a consul cluster for connect injection tests.
+type ConnectHelper struct {
+	// ClusterKind is the kind of Consul cluster to use (e.g. "Helm", "CLI").
+	ClusterKind consul.ClusterKind
 
-		"global.tls.enabled":           strconv.FormatBool(secure),
-		"global.tls.enableAutoEncrypt": strconv.FormatBool(autoEncrypt),
-		"global.acls.manageSystemACLs": strconv.FormatBool(secure),
-	}
+	// Secure determines if the cluster is considered "secure" based on the Helm values configured.
+	Secure bool
 
-	releaseName := helpers.RandomName()
-	var consulCluster consul.Cluster
-	if cli {
-		consulCluster = consul.NewCLICluster(t, helmValues, ctx, cfg, consul.CLIReleaseName)
-	} else {
-		consulCluster = consul.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+	// AutoEncrypt determines if the cluster is configured to use auto-encrypt.
+	AutoEncrypt bool
+
+	// HelmValues are the additional helm values to use when installing or upgrading the cluster beyond
+	// connectInject.enabled, global.tls.enabled, global.tls.enableAutoEncrypt, global.acls.mangageSystemACLs
+	// which are set by the Secure and AutoEncrypt fields.
+	HelmValues map[string]string
+
+	// RelaseName is the name of the cluster that will be generated by the ClusterGenerator.
+	ReleaseName string
+
+	T   *testing.T
+	Ctx environment.TestContext
+	Cfg *config.TestConfig
+
+	// consulCluster is the cluster that was generated by the ClusterGenerator.
+	consulCluster consul.Cluster
+
+	// consulClient is the consul client that will be used to test service mesh connectivity.
+	consulClient *api.Client
+}
+
+// Setup creates a new cluster using the New*Cluster function and assigns the cluster to the consulCluster field.
+func (c *ConnectHelper) Setup() {
+	switch c.ClusterKind {
+	case consul.Helm:
+		c.consulCluster = consul.NewHelmCluster(c.T, c.helmValues(), c.Ctx, c.Cfg, c.ReleaseName)
+	case consul.CLI:
+		c.consulCluster = consul.NewCLICluster(c.T, c.helmValues(), c.Ctx, c.Cfg, c.ReleaseName)
 	}
+}
 
-	consulCluster.Create(t)
+// Install creates a new cluster using the ClusterGenerator function then runs its Create method
```

Comment:
What's the ClusterGenerator this is referring to? Was just wondering if the name changed!